File: Proxy/Golang/ProxyNetwork.go
package main

import "fmt"

type ProxyNetwork struct {
	network INetwork
	bannedSites []string
}

func NewProxyNetwork(network INetwork) *ProxyNetwork {
	return &ProxyNetwork{
		network: network,
		bannedSites: []string{
			"abc.com",
			"xyz.com",
		},
	}
}

func (proxyNetwork *ProxyNetwork) connect(sites string)  {
	for _, bannedSite := range proxyNetwork.bannedSites {
        if sites == bannedSite {
			fmt.Println("Sites", sites, "Blocked")
			return
        }
    }

	proxyNetwork.network.connect(sites)
}

File: Proxy/Python 3/ProxyNetwork.py
from INetwork import INetwork

class ProxyNetwork(INetwork):
  def __init__(self, network):
    super().__init__()
    self.network = network
    self.banned_sites = [
      "abc.com",
      "xyz.com"
    ]

  def connect(self, site):
    if site in self.banned_sites:
      print("Access to", site, "has been denied.")
    else:
      self.network.connect(site)